modules/navier_stokes/src/fvkernels/INSFVTurbulentDiffusion.C

```diff
@@ -54,12 +58,21 @@ INSFVTurbulentDiffusion::computeQpResidual()
   // If we are on internal faces, we interpolate the diffusivity as usual
   if (_var.isInternalFace(*_face_info))
   {
-    interpolate(_coeff_interp_method,
-                coeff,
-                _coeff(elemArg(), state),
-                _coeff(neighborArg(), state),
-                *_face_info,
-                true);
+    // If the diffusion coefficients are zero, then we can early return 0 (and avoid warnings if we
+    // have a harmonic interpolation)
+    const auto coeff_elem = _coeff(elemArg(), state);
+    const auto coeff_neighbor = _coeff(neighborArg(), state);
+    if (!coeff_elem.value() && !coeff_neighbor.value())
+    {
+      if (!_newton_solve)
+        return 0;
+      else
+        // we return 0 but preserve the sparsity pattern for Newton's method
+        return 0 * (coeff_elem + coeff_neighbor) *
+               (_scaling_coef(elemArg(), state) + _scaling_coef(neighborArg(), state)) * dudn;
```

Comment:
This is a little awkward in a user-facing code. Why don't we need this in the regular diffusion kernel? Is this a fix for the restart?

Reply:
we actually do from this PR onward, see FVDiffusion